### DISCUSSION TITLE
Getting an error:404 when trying to install following the latest instructions

### DISCUSSION THREAD

**Discussion by tiredandtested** [nodesource/distributions#1670] [Q&A] [answered]
I'm doing a new install on a Raspberry Pi 4 and I'm following the latest instructions for Debian based distributions.  I get as far as and enter to following line:

curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg

and it returns:

curl: (22) The requested URL returned error: 404
gpg: no valid OpenPGP data found.

I've tried it several times yesterday and today.  Am I doing something wrong?  Can I get the correct key somewhere else?
Many thanks in advance

**Comment by riosje** [comment]
this doesn't make sense.
I just tried to download it and it works.
try to open this URL in your browser, literally the file is there
 https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key
 
 Could you add more verbosity to the curl command, try to execute this, and post the output here
`curl -vvv -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key`

**Reply by tiredandtested** [reply]
Thanks for your help.  It didn't make sense to me either and I tried opening the url directly and got the same 404.  Here's the output of the command with more verbosity:
```
*   Trying 143.204.15.127:443...
* Connected to deb.nodesource.com (143.204.15.127) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.nodesource.com
*  start date: Feb 23 00:00:00 2023 GMT
*  expire date: Oct 11 23:59:59 2023 GMT
*  subjectAltName: host "deb.nodesource.com" matched cert's "*.nodesource.com"
*  issuer: C=US; O=Amazon; CN=Amazon RSA 2048 M01
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x558b209ce0)
> GET /gpgkey/nodesource-repo.gpg.key HTTP/2
> Host: deb.nodesource.com
> user-agent: curl/7.74.0
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* The requested URL returned error: 404
* stopped the pause stream!
* Connection #0 to host deb.nodesource.com left intact
curl: (22) The requested URL returned error: 404
```

**Reply by riosje** [reply]
OK I thing this is related to DNS cache or something like that.
Could you  execute `dig deb.nodesource.com` or in case you don't have `dig` installed you should have nslookup try to execute `nslookup deb.nodesource.com` and post the output here

**Reply by tiredandtested** [reply]
~ $ dig deb.nodesource.com

; <<>> DiG 9.16.44-Debian <<>> deb.nodesource.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 9703
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;deb.nodesource.com.            IN      A

;; ANSWER SECTION:
deb.nodesource.com.     0       IN      A       143.204.15.127

;; Query time: 3 msec
;; SERVER: 100.100.100.100#53(100.100.100.100)
;; WHEN: Wed Sep 27 15:12:26 BST 2023
;; MSG SIZE  rcvd: 63

**Reply by riosje** [reply, marked as answer]
try to change your nameservers(DNS)
you should be getting an output similar to this one
```
; <<>> DiG 9.10.6 <<>> deb.nodesource.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 47275
;; flags: qr rd ra; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;deb.nodesource.com.		IN	A

;; ANSWER SECTION:
deb.nodesource.com.	300	IN	CNAME	deb.nodesource.com.cdn.cloudflare.net.
deb.nodesource.com.cdn.cloudflare.net. 300 IN A	172.67.10.205
deb.nodesource.com.cdn.cloudflare.net. 300 IN A	104.22.4.26
deb.nodesource.com.cdn.cloudflare.net. 300 IN A	104.22.5.26
```

**Reply by tiredandtested** [reply]
Many thanks for your help.  It is now installed without any further issues.  In case anyone else comes across this, I tried changing the dns in my Raspberry Pi 4 that I was trying to install Nodejs on and the Raspberry Pi 3 I run Pi Hole on and it still didn't work.  Then I changed the dns settings in Tailscale that I was also running on the Raspberrry Pi 3 and that enabled me to download the key.

Of all the installs I've done on this Raspberry Pi4 https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key was the only url I had issues with.  Not sure why this site behaves differently compared to others, maybe someone can explain but the above worked. Once again thank  you.

**Reply by jbreda007** [reply]
I am having a similar problem which I cant resolve.  On a Pi 3 I have tried to update nodered and nodejs but it fails as it cant resolve https://deb.nodesource.com.  If I ping deb.nodesource.com I get no response and it shows an inc $ ping deb.nodesource.com

PING deb.nodesource.com (143.204.15.127) 56(84) bytes of data.

Yet if I use dig I get this:
dig deb.nodesource.com

; <<>> DiG 9.11.5-P4-5.1+deb10u11-Raspbian <<>> deb.nodesource.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 20073
;; flags: qr rd ra; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;deb.nodesource.com.		IN	A

;; ANSWER SECTION:
deb.nodesource.com.	140552	IN	CNAME	deb.nodesource.com.cdn.cloudflare.net.
deb.nodesource.com.cdn.cloudflare.net. 254 IN A	104.22.5.26
deb.nodesource.com.cdn.cloudflare.net. 254 IN A	172.67.10.205
deb.nodesource.com.cdn.cloudflare.net. 254 IN A	104.22.4.26

;; Query time: 11 msec
;; SERVER: 194.168.4.100#53(194.168.4.100)
;; WHEN: Mon Sep 02 08:40:31 BST 2024
;; MSG SIZE  rcvd: 146

I've changed DNS servers with no help.

host returns@
host deb.nodesource.com
deb.nodesource.com is an alias for deb.nodesource.com.cdn.cloudflare.net.
deb.nodesource.com.cdn.cloudflare.net has address 172.67.10.205
deb.nodesource.com.cdn.cloudflare.net has address 104.22.5.26
deb.nodesource.com.cdn.cloudflare.net has address 104.22.4.26
deb.nodesource.com.cdn.cloudflare.net has IPv6 address 2606:4700:10::ac43:acd
deb.nodesource.com.cdn.cloudflare.net has IPv6 address 2606:4700:10::6816:41a
deb.nodesource.com.cdn.cloudflare.net has IPv6 address 2606:4700:10::6816:51a

Driving me mad.
